tracker/index.js

```diff
@@ -73,17 +58,22 @@ import { removeTrailingSlash } from '../lib/url';
     return a;
   };
 
-  const collect = (type, payload) => {
+  const collect = (type, payload, persist = false) => {
     if (trackingDisabled()) return;
-
-    post(
-      `${root}/api/collect`,
-      {
-        type,
-        payload,
-      },
-      res => (cache = res),
-    );
+    const endpoint = `${root}/api/collect`;
+    let headers = { 'Content-Type': 'application/json' };
+    if (cache) headers['x-umami-cache'] = cache;
+    let options = {
+      method: 'POST',
+      body: JSON.stringify({type, payload}),
+      headers
+    };
+    if (!persist) {
+      fetch(endpoint, options).then(res => res.text()).then(resText => { cache = resText });
+    } else {
+      options['keepalive'] = true;
```

Comment:
I think, all the requests should use the `keepalive` option. Previously, the other requests were sent using XHR, therefore those requests were not persisted.